block-node/messaging/src/main/java/org/hiero/block/server/messaging/impl/MessagingServiceImpl.java

```diff
@@ -0,0 +1,300 @@
+
+                @Override
+                public void handleOnStartException(Throwable ex) {
+                    LOGGER.log(
+                            Level.ERROR, "Exception in block notification disruptor startup: " + ex.getMessage(), ex);
+                }
+
+                @Override
+                public void handleOnShutdownException(Throwable ex) {
+                    LOGGER.log(
+                            Level.ERROR, "Exception in block notification disruptor shutdown: " + ex.getMessage(), ex);
+                }
+            };
+
+    /**
+     * The disruptor that handles the block item batches. It is used to send block items to the different handlers.
+     * It is a single producer, multiple consumer disruptor.
+     */
+    private final Disruptor<BlockItemBatchRingEvent> blockItemDisruptor;
+
+    /**
+     * The disruptor that handles the block notifications. It is used to send block notifications to the different
+     * handlers. It is a single producer, multiple consumer disruptor.
+     */
+    private final Disruptor<BlockNotificationRingEvent> blockNotificationDisruptor;
+
+    /** Collect all block item handlers till start then register them with the disruptor. */
+    private final ArrayList<EventHandler<BlockItemBatchRingEvent>> blockItemHandlers = new ArrayList<>();
+
+    /** Collect all block notification handlers till start then register them with the disruptor. */
+    private final ArrayList<EventHandler<BlockNotificationRingEvent>> blockNotificationHandlers = new ArrayList<>();
+
+    /** Map of dynamic no back pressure block item handlers to their threads. So that we can stop them */
+    private final Map<NoBackPressureBlockItemHandler, Thread> dynamicNoBackPressureBlockItemHandlers = new HashMap<>();
+
+    /** Map of dynamic no back pressure block item handlers to their event processors. So that we can stop them */
+    private final Map<NoBackPressureBlockItemHandler, BatchEventProcessor<BlockItemBatchRingEvent>>
+            dynamicNoBackPressureBlockItemHandlerToEventProcessor = new HashMap<>();
+
+    /**
+     * Constructs a new MessagingServiceImpl instance with the default configuration. It uses the
+     * ConfigurationBuilderFactory to load the configuration from the classpath.
+     */
+    public MessagingServiceImpl() {
+        this(getConfig());
+    }
+
+    /**
+     * Constructs a new MessagingServiceImpl instance with the given configuration.
+     *
+     * @param config the configuration for the messaging service
+     */
+    public MessagingServiceImpl(final MessagingConfig config) {
+        blockItemDisruptor = new Disruptor<>(
+                BlockItemBatchRingEvent::new,
+                config.queueSize(),
+                VIRTUAL_THREAD_FACTORY,
+                ProducerType.SINGLE,
+                new SleepingWaitStrategy());
+        blockNotificationDisruptor = new Disruptor<>(
+                BlockNotificationRingEvent::new,
+                config.queueSize(),
```

Comment:
Q: should we have different queueSizes for the two ring buffers. As is changing one as part of a perf optimization would modify the other.

Reply:
I am trying to avoid config unless absolutely necessary. We have so much config in consensus node and have no idea what to do with it. Most config is just guesses or set and pray, not actually needed.  Most of these settings should be decided based on testing and just hard code in. Only if a setting needs to be configured at runtime by a ops person should it be exposed as config. If we made everything config I could see us having 1000s of config params, can you even imagine trying to document what they all do and best practices for setting them. We should discuss the philosophy with the team.

In this case 1024 should be plenty, if we have issues in performance testing maybe we will tweak it once or twice and then should be happy.

Reply:
If everything is well designed then this setting should not be important. In almost all cases we should be within a couple items of the head of the ring buffer.